code/datums/mapgen/biomes/_biome.dm

```diff
@@ -0,0 +1,60 @@
+///This datum handles the transitioning from a turf to a specific biome, and handles spawning decorative structures and mobs.
+/datum/biome
+	///Type of turf this biome creates
+	var/turf_type
+	///Chance of having a structure from the flora types list spawn
+	var/flora_density = 0
+	///Chance of having a mob from the fauna types list spawn
+	var/fauna_density = 0
+	///list of objects that can be spawned when the turf spawns flora
+	var/list/flora_types = list(/obj/structure/flora/grass/jungle)
+	///list of mobs that can be spawned when the turf spawns fauna
+	var/list/fauna_types = list(
+	/mob/living/simple_animal/hostile/jungle/mook{faction = list("wildlife")} = 50,
+	)
+
+///This proc handles the creation of a turf of a specific biome type
+/datum/biome/proc/generate_turf(var/turf/T)
+	T.ChangeTurf(turf_type)
```

Comment:
You probably want to turn off AfterChange processing on the generation for the whole block in case this is done post-roundstart, ie aways and such.